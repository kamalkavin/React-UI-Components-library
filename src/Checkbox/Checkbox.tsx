/**
 *
 * Checkbox
 *
 */

import React, { FC } from 'react';
import { FormCheck, FormCheckProps } from 'react-bootstrap';
import { UseFormMethods, ValidationRules } from 'react-hook-form';
import cx from 'classnames';

interface FormProps extends Partial<Pick<UseFormMethods, 'register' | 'errors'>> {
	rules?: ValidationRules;
	name: string;
	label: string;
	variant?: 'small' | 'medium' | 'large';
}

export interface CheckboxProps extends Omit<FormCheckProps, 'name'>, Omit<FormProps, 'label'> {}

const Checkbox: FC<CheckboxProps> = (props: CheckboxProps) => {
	const variant = props.variant || 'small';
	return (
		<>
			{props.register ? (
				<>
					<FormCheck
						className={cx('input', props.errors && props.errors[props.name] && 'is-danger', variant)}
						aria-invalid={props.errors && props.errors[props.name] ? 'true' : 'false'}
						type={props.type}
						data-cy={props.name}
						id={props.name}
						ref={props.register && props.register(props.rules)}
						label={props.label}
						{...props}
					/>
					{props.errors && props.errors[props.name] && (
						<div>
							<span data-cy='checkbox-error' className='text-danger'>
								{props.errors && props.errors[props.name].message}
							</span>
						</div>
					)}
				</>
			) : (
				<FormCheck
					className={cx('input', props.errors && props.errors[props.name] && 'is-danger', variant)}
					aria-invalid={props.errors && props.errors[props.name] ? 'true' : 'false'}
					type={props.type}
					data-cy={props.name}
					id={props.name}
					label={props.label}
					{...props}
				/>
			)}
		</>
	);
};

export default Checkbox;
