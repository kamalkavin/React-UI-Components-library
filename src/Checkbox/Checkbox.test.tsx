/**
 *
 * Tests for Checkbox
 *
 *
 */

import React from 'react';
import Checkbox, { CheckboxProps } from './Checkbox';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
describe('Checkbox Component', () => {
	afterEach(() => {
		cleanup();
	});
	test('Checkbox test case', () => {
		const props: CheckboxProps = {
			name: 'test',
			type: 'checkbox',
			className: 'test-class',
			label: 'on',
			errors: {
				test: {
					message: 'Requird',
				},
			},
		};
		render(<Checkbox {...props} />);
		expect(screen.getByTestId(props.name)).toBeInTheDocument();
		fireEvent.input(screen.getByTestId(props.name), {
			target: {
				value: true,
			},
		});
	});
});
