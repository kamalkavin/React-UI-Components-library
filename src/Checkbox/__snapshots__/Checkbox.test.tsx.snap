// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshot renders correctly 1`] = `
<Fragment>
  <FormCheck
    aria-invalid="false"
    className="input"
    data-cy=""
    id=""
    name=""
  />
</Fragment>
`;
